src/pages/index.js

```diff
@@ -7,96 +7,208 @@ import Card from "../components/Card.js";
 /* ---------------------------------------------------------------------------------------------- */
 
-const userInfo = new UserInfo(profileTitleSelector, profileDescriptionSelector);
+editProfileFormValidator.enableValidation();
+addCardFormValidator.enableValidation();
+avatarFromValidator.enableValidation();
+deleteCardFormValidator.enableValidation();
+/* ---------------------------------------------------------------------------------------------- */
+/*                                           API Promise                                          */
+/* ---------------------------------------------------------------------------------------------- */
 
-const editProfilePopup = new PopupWithForm(profileEditModalSelector, (inputsObject) => {
-  userInfo.setUserInfo(inputsObject.title, inputsObject.description);
-  editProfilePopup.close();
-});
+let userId;
+
+//! Do Not Delete
+Promise.all([api.getInitialCards(), api.getUserInfo()])
+  .then(([initialCards, userData]) => {
+    userId = userData._id;
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setProfileAvatar(userData.avatar);
+    const cardListSection = new Section(
+      {
+        items: initialCards,
+        renderer: (data) => {
+          const newCard = createCard(data);
+          cardListSection.addItem(newCard);
+        },
+      },
+      cardListElement
+    );
+    cardListSection.renderItems();
+  })
+  .catch(() => (err) => console.log(err));
 
-profileEditButton.addEventListener("click", openProfilePopup);
+/* ---------------------------------------------------------------------------------------------- */
+/*                                        Profile Functions                                       */
+/* ---------------------------------------------------------------------------------------------- */
 
+//! Do Not Delete
 function openProfilePopup() {
   const { profileName, description } = userInfo.getUserInfo();
   profileTitleElement.value = profileName;
   profileDescriptionElement.value = description;
   editProfileFormValidator.toggleButtonState();
   editProfilePopup.open();
+  editProfilePopup.setLoading(false, "Save");
 }
 
-/* ---------------------------------------------------------------------------------------------- */
-/*                                          Section Class                                         */
-/* ---------------------------------------------------------------------------------------------- */
-
-const cardListSection = new Section(
-  {
-    items: initialCards,
-    renderer: ({ name, link }) => {
-      const newCard = createCard({ name, link });
-      cardListSection.addItem(newCard);
-    },
-  },
-  cardList
-);
-
-cardListSection.renderItems();
+//! Do Not Delete
+function handleProfileFormSubmit({ title, description }) {
+  api
+    .updateUserInfo(title, description)
+    .then(() => {
+      userInfo.setUserInfo(title, description);
+      editProfilePopup.setLoading(true);
```

Comment:
`.setLoading(true)` should be called **before** sending requests to see the button text changing